File: api/src/schemas/notification.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel

class NotificationRead(BaseModel):
    uuid: str
    user_uuid: str
    type: str
    content: str
    read: bool

    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

class NotificationCreate(BaseModel):
    user_uuid: str
    type: str
    content: str

class NotificationUpdate(BaseModel):
    content: Optional[str] = None
    read: Optional[bool] = None

File: frontend/src/models/Notification.js
export default class Notification {
    constructor({
        uuid = '',
        user_uuid = '',
        type = '',
        content = '',
        read = true,
        created_at = '',
        updated_at = '',
    }) {
        this.uuid = uuid;
        this.user_uuid = user_uuid;
        this.type = type;
        this.content = content;
        this.read = read;
        this.created_at = created_at;
        this.updated_at = updated_at;
    }
}